tox.ini

```diff
@@ -62,18 +66,19 @@ commands =
 [pinned]
 deps =
     -ctests/constraints.txt
-    cryptography==2.0
+    #using cryptography-2.3.1 to solve dependencies conflict on pip20.2>=
+    cryptography==2.3.1
     cssselect==0.9.1
     itemadapter==0.1.0
     parsel==1.5.0
     Protego==0.1.15
-    PyDispatcher==2.0.5
-    pyOpenSSL==16.2.0
+    #using pyOpenSSL<18.1 to solve dependencies conflict with mitmproxy on pip20.2>=
+    pyOpenSSL==18.0.0
     queuelib==1.4.2
     service_identity==16.0.0
-    Twisted==17.9.0
+    Twisted==20.3.0
     w3lib==1.17.0
-    zope.interface==4.1.3
+    zope.interface==5.1.2
```

Comment:
Why is this needed?

Reply:
To resolve dependencies conflict with Twisted. Probably should've left a comment there